src/routes.ts

```diff
@@ -39,21 +39,21 @@ export interface RouteMethod {
 }
 
 export interface RouteHandler {
-  (page: PageBase, model: Model, params: PageParams, next: (err?: Error) => void): void;
+  (page: Page, model: RootModel, params: PageParams, next: (err?: Error) => void): void;
 }
 
 export interface TransitionalRouteHandler {
   (
-    page: PageBase,
-    model: Model,
+    page: Page,
+    model: RootModel,
     params: PageParams,
     next: (err?: Error) => void,
     done: () => void
   ): void;
 }
 
 declare module './App' {
-  interface AppBase {
+  interface App {
     del: RouteMethod;
     get: RouteMethod;
     history: {
```

Comment:
per discussion, will leave this as is for now due to large number of places in production code that this would need to be cast for more "correct" typing. Makes for worse DX so we will forgo changing for now